packages/p2p/src/stores/my-profile-store.js

```diff
@@ -18,14 +19,18 @@ export default class MyProfileStore extends BaseStore {
     @observable delete_error_message = '';
     @observable error_message = '';
     @observable form_error = '';
+    @observable formik_ref = null; // used to keep track of formik's ref when updated
+    @observable formik_history = null; // used to store formik's ref when unmounting
     @observable full_name = '';
     @observable is_button_loading = false;
     @observable is_cancel_add_payment_method_modal_open = false;
     @observable is_cancel_edit_payment_method_modal_open = false;
     @observable is_confirm_delete_modal_open = false;
     @observable is_delete_payment_method_error_modal_open = false;
     @observable is_loading = true;
+    @observable is_modal_transitioning = false;
```

Comment:
Why do we need this?

Reply:
Its to avoid re-fetching the adverts every time `BuySellModal` is not visible only during a modal transition (i.e. showing an error modal or cancel confirmation modal like `CancelAddPaymentMethodModal`) in the background, I added comment about this in `buy-sell-store` in line 458. This observable is to keep track and identify modal transitions like these